log4j-kit/src/main/java/org/apache/logging/log4j/kit/env/PropertySource.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.kit.env;
+
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Basic interface to retrieve property values.
+ * <p>
+ *     We can not reuse the property sources from 2.x, since those required some sort of {@code log4j} prefix to be
+ *     included. In 3.x we want to use keys without a prefix.
+ * </p>
+ */
+public interface PropertySource {
+    /**
+     * Provides the priority of the property source.
+     * <p>
+     *     Property sources are ordered according to the natural ordering of their priority. Sources with lower
+     *     numerical value take precedence over those with higher numerical value.
+     * </p>
+     *
+     * @return priority value
+     */
+    int getPriority();
```

Comment:
There is no coherence here in Log4j or other Java projects:

- Spring [`@Order`](https://www.baeldung.com/spring-order), Log4j's old [`PropertySource`](https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/util/PropertySource.html#getPriority()) and [`Provider`](https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/spi/Provider.html#getPriority()) use a **lowest-first** ordering,
- Log4j's [`@Order`](https://logging.apache.org/log4j/2.x/javadoc/log4j-core/org/apache/logging/log4j/core/config/Order.html) annotation uses a **highest-first** ordering.

Reply:
I agree.

Note that `@Order` does not use the word `priority`. Hence, I don't think it is a valid example to compare with.

Regarding `PropertySource` and `Provider`, IMHO, they got it wrong. :stuck_out_tongue_closed_eyes: I will consult to native speakers in Slack...

Reply:
Well, a lot of this is my fault and I am a native speaker.  The problem starts with DEBUG, INFO, WARN, and ERROR.  Many people want to discuss them as one Level being "higher" or "lower" than another. However, the whether a Level is "higher" or "lower" differs based on whether you are viewing it for the develoepr writing code or from the configuration.  In the code, debug would typically be thought of low level logging. But when you set the log level to DEBUG in the configuration you are capturing MORE logging data, which could be perceived as being higher.  That is why I avoided that terminology completely when discussing levels and used less specific and more specific. 

Ordering has the same problem. When something has the highest value does that mean it is processed first or last? You could argue it either way. It also has to do with how defaults get handled. For example, if the item with the highest value is processed first what happens when you get a duplicate entry on an item with a lower value. It could either replace the previous item or be ignored by the previous item, depending on which item should take precedence.

In short, I don't think there is any kind of standard for this.

Reply:
@ppkarwasz, I don't want to this conversation to block your PR. You can proceed as you wish. Please share your conclusion here and resolve this conversation at your convenience.